stwcs/updatewcs/__init__.py

```diff
@@ -109,17 +109,13 @@ def updatewcs(input, vacorr=True, tddcorr=True, npolcorr=True, d2imcorr=True,
         for item in file_names:
             files.append(fits.open(item, mode='update'))
 
-    if checkfiles:
-        files = checkFiles(files)
-        if not files:
-            print('No valid input, quitting ...\n')
-            return
     logger.info("\n\tInput files: {}".format(file_names))
     logger.info("\n\tInput arguments: %s" % args)
 
-    print('names', file_names)
+    print('file_names', file_names)
     if checkfiles:
         files = checkFiles(files)
+        file_names = [inp.filename() for inp in files]
```

Comment:
I think this may fail in some cases. For example, if the input was a wavered fits. In case it's not written to file the `fileopbj.filename()` returns None.I have a PR with changes for the wavered fits working with HDUList objects but need more time to test it before submitting it.

Reply:
The test I have added starts with a waivered FITS WFPC2 file, and it works just fine with this code.

Reply:
Actually, this was producing an error for STIS ASN files, since the separate exposure MEF files generated from the ASN were not originally being passed back as HDULists that were associated with a file on disk.  I have addressed this problem with [stsci.tools PR #90](https://github.com/spacetelescope/stsci.tools/pull/90).  In addition, the new tests added for updatewcs start with a waivered FITS WFPC2 file and a STIS ASN file, both of which pass with the stsci.tools PR in place (at least in my dev area).